supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchJob.scala

```diff
@@ -195,32 +187,21 @@ final case class AwsBatchJob(jobDescriptor: BackendJobDescriptor,           // W
                               .`type`(JobDefinitionType.CONTAINER)
                               .build
 
-    var arn = "" // Use of null here due to underlying submitJob call
-    for (inx <- 1 to 6) {
-      try {
-        if (arn == "") {
-          arn = client.registerJobDefinition(definitionRequest).jobDefinitionArn
-        }
-      } catch {
-        case e: ClientException if e.statusCode() == 404 && inx < 6 => {
-          // RegisterJobDefinition throws 404s every once in a while
-          Log.warn(s"RegisterJobDefinition 404 - retrying (${inx}/5)")
-          Thread.sleep(100L * (inx ^ 2L))
-        }
-        case e: ClientException if e.statusCode() == 409 => {
-          // This could be a problem here, as we might have registerJobDefinition
-          // but not describeJobDefinitions permissions. We've changed this to a
-          // warning as a result of this potential
-          Log.warn("Job definition already exists. Performing describe and retrieving latest revision")
-          val defs = client
-            .describeJobDefinitions(DescribeJobDefinitionsRequest.builder().jobDefinitionName(sanitize(name)).build())
-            .jobDefinitions()
-          // We need latest revision
-          defs.get(defs.size() - 1).jobDefinitionArn()
-        }
-      }
+    val submit = async.delay(client.registerJobDefinition(definitionRequest).jobDefinitionArn)
+
+    val retry: F[String] = Scheduler.fromScheduledExecutorService(ses).retry(submit, 0.millis, _ * 2, 6, {
+      case e: ClientException => e.statusCode() == 404
+      case _ => false
+    }).compile.last.map(_.get)
+
+    async.recoverWith(retry) {
+      case e: ClientException if e.statusCode() == 409 =>
+        Log.warn("Job definition already exists. Performing describe and retrieving latest revision")
```

Comment:
Is it something that's expected to happen frequently ? For instance if we scatter 10000 wide would we try to create the same job definition 10000 times and so log this message 9999 time ? If so I'd advocate for not logging this (or not at warn level at least)

Reply:
It's a permissions thing, per comments above:
```scala
          // This could be a problem here, as we might have registerJobDefinition
          // but not describeJobDefinitions permissions. We've changed this to a
          // warning as a result of this potential
```
Thus we eitehr get 10,000 of them or we get 0.  I think `WARN` is ok IMO

Reply:
True... I don't feel strongly about this issue as it is an edge case.  I'll put it in as `INFO` unless I hear a vote in favor of `DEBUG`

Reply:
If this really is a "normal part of operation that will happen every time we scatter" I'd say don't log it at all (at least not separate messages for *every* *single* *shard*).

Reply:
Per the original author's comment (I've since added back), it's clear that there is a potential issue w/ this logic and warrants a `WARN`.  I'm going to honor these intentions and leave it as a `WARN`.